### DISCUSSION TITLE
Change UML attributes and methods

### DISCUSSION THREAD

**Discussion by aylafiryal** [clientIO/joint#2220] [Q&A] [answered]
### Introduction

Is it possible to change the attributes and methods of UML shapes in real-time? I've been trying to do it like this, but it doesn't seem to have any effect. I've attempted both through the console and on the paper:
`var id = this.model.id;`
`var el = graph.getCell(id);`
`el.attr({`
     `attributes: atr,`
`});`

Should I use [element event](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.Element.events)? But honestly, I'm still a bit confused about using this. I would greatly appreciate further explanation or additional examples!

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment, marked as answer]
Use the [prop()](https://resources.jointjs.com/docs/jointjs/v3.7/joint.html#dia.Element.prototype.prop) instead.


The following two expressions are equivalent:
```js
el.prop('attrs/body/fill', 'red');
el.attr('body/fill', 'red');
```

The `attributes` attribute of the UML shape is a top-level attribute, hence:

```js
umlEl.prop('attributes', myAttributes);
```

Note that if the `attributes` were arrays of objects rather than primitives (strings), you would have to pass the `rewrite` flag to prevent merging.

```js
el.prop('myArrayWithObjects', [{ name: 'a', a: 1 }, { name: 'b', b: 1 }], { rewrite: true });
```

**Reply by aylafiryal** [reply]
I'm so sorry for the late response. Thank you so much for your help! This works for me
